frontend/src/Pages/SearchPage.tsx

```diff
@@ -101,7 +101,7 @@ export default function SearchPage() {
         } else {
             setFocus('term');
         }
-    }, [defaultSearch]);
+    }, []);
```

Comment:
I think this is a backward way of fixing this. Generally `useEffect` should have all variables, that are used inside, as its dependencies.

Instead of removing `defaultSearch` from the dependencies list, I suggest calling `setSearchParams({term});` on form submit, instead of calling `doSearch` directly.

Reply:
No, I'm pretty sure removing dependency is the right fix for this use case. The logic should be applied only once on the page load (physical page load, not component render)